src/lib/convert-forms/index.js

```diff
@@ -0,0 +1,183 @@
+const path = require('node:path');
+const { DOMParser, XMLSerializer } = require('@xmldom/xmldom');
+const xmlFormat = require('xml-formatter');
+const { replaceBase64ImageDynamicDefaults, replaceItemSetsWithMedia } = require('./handle-media');
+const { removeNoLabelNodes } = require('./handle-no-label-placeholders');
+const { removeExtraRepeatInstance, addRepeatCount } = require('./handle-repeat');
+const { handleDbDocRefs } = require('./handle-db-doc-ref');
+
+const domParser = new DOMParser();
+const serializer = new XMLSerializer();
+const XLS2XFORM = path.join(__dirname, '..', '..', '..', 'bin', 'xls2xform-medic');
+
+const FORM_EXTENSION = '.xlsx';
+const formFileMatcher = (fileName) => {
+  if (fileName.endsWith(FORM_EXTENSION) &&
+    !fileName.startsWith('~$') // ignore Excel "owner files"
+    && fileName !== 'PLACE_TYPE-create.xlsx' && fileName !== 'PLACE_TYPE-edit.xlsx') {
+    return fileName.slice(0, fileName.length - FORM_EXTENSION.length);
+  }
+  return null;
+};
+
+const execute = async (projectDir, subDirectory, options = {}) => {
+  const formsDir = getFormDir(projectDir, subDirectory);
+
+  if (!fs.exists(formsDir)) {
+    warn(`Forms dir not found: ${formsDir}`);
+    return;
+  }
+
+  const filesToConvert = argsFormFilter(formsDir, FORM_EXTENSION, options)
+    .filter(name => formFileMatcher(name));
+
+  for (const xls of filesToConvert) {
+    const originalSourcePath = `${formsDir}/${xls}`;
+    let sourcePath;
+
+    if (options.force_data_node) {
+      const temporaryPath = `${fs.mkdtemp()}/${options.force_data_node}.xlsx`;
+      fs.copy(originalSourcePath, temporaryPath);
+      sourcePath = temporaryPath;
+    } else {
+      sourcePath = originalSourcePath;
+    }
+
+    const targetPath = `${fs.withoutExtension(originalSourcePath)}.xml`;
+
+    info('Converting form', originalSourcePath, '…');
+
+    await xls2xform(escapeWhitespacesInPath(sourcePath), escapeWhitespacesInPath(targetPath), xls);
+    const hiddenFields = await getHiddenFields(`${fs.withoutExtension(originalSourcePath)}.properties.json`);
+    fixXml(targetPath, hiddenFields, options.transformer, options.enketo);
+    trace('Converted form', originalSourcePath);
+  }
+};
+
+module.exports = {
+  SUPPORTED_EXTENSIONS: [FORM_EXTENSION],
+  formFileMatcher,
+  execute
+};
+
+const PYXFORM_EMPTY_GROUP_ERROR = '\'NoneType\' object is not iterable';
+const PYXFORM_CODES = {
+  OK: 100,
+  WARNING: 101,
+  // ERROR: 999
+};
+
+const pyxformErrorMessage = (e) => {
+  const errorMsg = typeof e === `string` ? e : e.message || JSON.stringify(e);
+  if (PYXFORM_EMPTY_GROUP_ERROR === errorMsg) {
+    // Pyxform does not gracefully handle empty groups https://github.com/XLSForm/pyxform/issues/754
+    return `Check the form for an empty group or repeat.`;
+  }
+  return errorMsg;
+};
+
+const xls2xform = async (sourcePath, targetPath, xlsxFileName) => {
+  const result = await exec([XLS2XFORM, '--skip_validate', '--json', sourcePath, targetPath], LEVEL_NONE)
```

Comment:
Setting `--json` here causes pyxform to emit the results of the conversion (success, error, warning, etc) as a JSON object that we can parse programmatically.

This is nice because instead of dumping a bunch of pyxform stacktraces to the console when form conversion fails, we can give just a clear error message.